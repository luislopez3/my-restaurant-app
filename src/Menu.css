.menu-by-type {
  background-color: #ffffef !important;
  font-size: 13px;
  color: #56593d;
  text-align: left;
  padding-left: 20px;
}

.menu-img {
  width: 21rem;
  height: 21rem;
}

.full-menu-row {
  background-color: #ffffef;
  display: inline-block;
  width: 23rem;
  line-height: 20px;
}

.full-menu-column {
  background-color: #ffffef;
  width: 92%;
  text-align: left;
  padding-left: 15px;
  margin-top: 25px;
}

.add-to-order-button {
  background-color: #56593d;
  color: #f8f0e2;
  font-weight: bold;
  width: 8rem;
  padding: 8px;
}

/* -------  Media Query ------- */

@media all and (max-width: 768px) {
  .menu-img {
    width: 16rem;
    height: 16rem;
  }

  .full-menu-row {
    display: inline-block;
    font-size: 12px;
    width: 95%;
  }

  .full-menu-column {
    width: 72%;
    margin-top: 10px;
  }

  .add-to-order-button {
    width: 5rem;
    padding: 5px;
    margin-top: 10px;
  }
}
