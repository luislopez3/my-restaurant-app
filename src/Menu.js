import React, { useEffect, useState, useContext } from "react";
import { useParams } from "react-router-dom";
import AppContext from "./AppContext";

export default function Menu() {
  const [menus, setMenus] = useState([]);
  const [bag, setBag] = useState([]);
  const { type } = useParams();
  const { order_id, setOrderId } = useContext(AppContext);

  useEffect(() => {
    fetch(`http://localhost:5000/menu_items?type=${type}`)
      .then((response) => {
        return response.json();
      })
      .then((data) => {
        setMenus(data);
      });
  }, [type]);

  console.log(menus, bag);

  function addToBag(menu) {
    fetch("http://localhost:5000/order_items", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ ...menu, order_id }),
    })
      .then((response) => {
        return response.json();
      })
      .then((data) => {
        setBag(data);
        if (!order_id) {
          setOrderId(data.order_id);
          localStorage.order_id = data.order_id;
        }
      });
  }

  return (
    <>
      {menus.map((menu, index) => {
        return (
          <ul key={index}>
            <p>{menu.name}</p>
            <p>{menu.description}</p>
            <img src={menu.image_url} alt={menu.name} />
            <p>
              Price:{" "}
              {new Intl.NumberFormat("en-US", {
                style: "currency",
                currency: "USD",
              }).format(menu.price)}
            </p>
            <button onClick={() => addToBag(menu)}>Add to Bag</button>
          </ul>
        );
      })}
    </>
  );
}
