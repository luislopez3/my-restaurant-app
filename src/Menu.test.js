import React from "react";
import { MemoryRouter } from "react-router-dom";
import { create } from "react-test-renderer";
import Menu from "./Menu";
import { render } from "@testing-library/react";

describe("Menu component", () => {
  test("renders Menu w/o an error", () => {
    render(
      <MemoryRouter>
        <Menu />
      </MemoryRouter>
    );
  });

  test("Matches the snapshot", () => {
    const wrapper = create(
      <MemoryRouter>
        <Menu />
      </MemoryRouter>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
