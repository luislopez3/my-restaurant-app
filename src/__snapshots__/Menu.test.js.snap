// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component Matches the snapshot 1`] = `null`;
